### ISSUE TITLE
How to iterate through native image resources directory

### ISSUE THREAD

**Issue by alexander-shustanov** [oracle/graal#1146]
If I have a jar file, I can get any packed resource directory as a Path object, and iterate all it files and subdirectories. Is there any way to do the same in native images?

The example on Kotlin:

```kotlin
clazz.getResource(resourceName)
                    ?.toURI()
                    ?.let {
                        if (it.scheme == "jar") {
                            val fileSystem = getFileSystem(it)
                            fileSystem.getPath(resourceName)
                        } else {
                            Paths.get(it)
                        }
                    }
```

**Comment by alexander-shustanov** [comment]
Now, I created my own implementation of FileSystem and provide all resources index.

**Comment by anusse** [comment]
Was this issue ever resolved?  Is there any way to include directory entries as resource entries in the native image?  I'm dealing with something similar:

I'm using some existing code (Liquibase) that attempts to load a directory path resource and then subsequently loads all the class files it can find from this path. I've also created a resource configuration file specifying the directories and files in this package that should be included when I generate the native image.  The native image can be successfully generated.  

However, when I run the executable, it fails to get a URL for the intermediate directory that I specified in the resource configuration file. However, it is able to find the files specified in the configuration file.

```
      // returns null
      clazz.getResource("com/example/animals/");

      // returns correct URL
      clazz.getResource("com/example/animals/example.txt");
```
